src/keys.js

```diff
@@ -354,6 +360,14 @@ export function validatePublicKey(pubkeyHex) {
     return "Invalid public key.";
   }
 
+  if (
+    !isKeyCompressed(pubkeyHex) &&
+    addressType &&
```

Comment:
I don't think you need the check for `addressType` since the next line will return false if `addressType` is undefined

Reply:
true!